src/FreeAge/server/game.cpp

```diff
@@ -1379,17 +1385,12 @@ bool Game::SimulateMeleeAttack(u32 /*unitId*/, ServerUnit* unit, u32 targetId, S
       timeSinceActionStart - stepLengthInSeconds < attackDamageTime &&
       target != nullptr) {
     // Compute the attack damage.
+    // TODO: handle all the attack types
+    CHECK_EQ(unit->GetStats().attackType == AttackType::Default, true);
```

Comment:
Yes, CHECK_EQ does not work with enum classes, I was looking for CHECK_TRUE but I guess it's just CHECK.

It looks like CHECK_EQ cannot compile because the `operator<<` is missing from the AttackType. I have added a TODO suggestion comment (somewhere) about implementing a GetXInternaName where X all the enums (mainly for debugging and logging). With that the `operator<<` would be trivial to implement.

Reply:
Alternatively (with less pretty output though), one can just do `static_cast<int>()`.